sql/catalyst/src/main/scala/org/apache/spark/sql/types/NullType.scala

```diff
@@ -32,6 +32,8 @@ class NullType private() extends DataType {
   override def defaultSize: Int = 1
 
   private[spark] override def asNullable: NullType = this
+
+  override def simpleString: String = "void"
```

Comment:
This is also an unrelated change. I like the new naming, but it's unnecessary to the hive compatibility issue.